src/Liquifier.sol

```diff
@@ -277,10 +287,14 @@ contract Liquifier is Initializable, UUPSUpgradeable, OwnableUpgradeable, Pausab
     // Send the redeemed ETH back to the liquidity pool & Send the fee to Treasury
     function withdrawEther() external {
         if (!roleRegistry.hasRole(LIQUIFIER_ADMIN_ROLE, msg.sender)) revert IncorrectRole();
+        _withdrawEther();
+    }
 
-        uint256 amountToLiquidityPool = address(this).balance;
-        (bool sent, ) = payable(address(liquidityPool)).call{value: amountToLiquidityPool, gas: 20000}("");
-        if (!sent) revert EthTransferFailed();
+    // Swap Liquifier's eETH for ETH from the liquidity pool and send it back to the liquidity pool
+    function withdrawEEth(uint256 amount) external {
+        if (!roleRegistry.hasRole(LIQUIFIER_ADMIN_ROLE, msg.sender)) revert IncorrectRole();
+        liquidityPool.withdraw(address(this), amount);
```

Comment:
You could change this and remove the need for the sequential  `_withdrawEther` call
```
liquidityPool.withdraw(address(liquidityPool), amount);
```

Reply:
Nice suggestion! Updated as suggested. Needed to update `LiquidityPool.withdraw` to handle the case when it is sending ETH to itself.